web-console/src/components/filler.tsx

```diff
@@ -257,3 +259,63 @@ export class TagInput extends React.Component<TagInputProps, { stringValue: stri
     />;
   }
 }
+
+interface JSONInputProps extends React.Props<any>{
+  onChange: (newValue: string) => void;
+  value: JSON;
+  updateErrors: ((newValue: string) => void)
```

Comment:
why the extra `()` ?

Reply:
What is the logic behind this function? Why does it take a string value and not an error?

Reply:
The dialog needs to know which JSON input has invalid value so the submit button could be disabled. It's taking a string since it's the raw value in the input; this raw value will be checked if it could be parsed into JSON. The dialog will then check if all fields have valid JSONs

Reply:
I am refactoring the entire JSON validating process. 

Originally, the dialog has an object where the property is the JSON input name. and value is a boolean to tell if the input is valid. The dialog gotta check if every property is true.

I'm changing this errors tracking/checking process into auto-form, and auto form will update dialog by telling a boolean value indicating if all JSON inputs are valid or not. Now, auto-form will have a state which is object {string: boolean}, tracking if every JSON input is valid. JSON input will do the checking and change the object in auto-form by setting boolean values. I think in this way it makes the auto form API easy to use when auto-form handles the tracking and JSON input handles validating.